tests/components/abode/test_lock.py

```diff
@@ -0,0 +1,62 @@
+"""Tests for the Abode lock device."""
+from unittest.mock import patch
+
+from homeassistant.components.abode import ATTR_DEVICE_ID
+from homeassistant.components.lock import DOMAIN as LOCK_DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    SERVICE_LOCK,
+    SERVICE_UNLOCK,
+    STATE_LOCKED,
+)
+
+from .common import setup_platform
+
+DEVICE_ID = "lock.test_lock"
+
+
+async def test_entity_registry(hass):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, LOCK_DOMAIN)
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entry = entity_registry.async_get(DEVICE_ID)
+    assert entry.unique_id == "51cab3b545d2o34ed7fz02731bda5324"
+
+
+async def test_attributes(hass):
+    """Test the lock attributes are correct."""
+    await setup_platform(hass, LOCK_DOMAIN)
+
+    state = hass.states.get(DEVICE_ID)
+    assert state.state == STATE_LOCKED
+    assert state.attributes.get(ATTR_DEVICE_ID) == "ZW:00000004"
+    assert not state.attributes.get("battery_low")
+    assert not state.attributes.get("no_response")
+    assert state.attributes.get("device_type") == "Door Lock"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5184102)